### PR TITLE
feat: Allow injectable transient storage

### LINKED ISSUE #672: Allow injectable transient storage
part of #622

### PR DESCRIPTION
**Pull request by fqutishat**
- Add mem store as default transient storage
- Allow user to inject custom transient storage

closes #672

### CODE REVIEW

File: pkg/storage/mem/mem_store.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	}
+	return nil
+}
+
+type memStore struct {
+	db map[string][]byte
+}
+
+// Put stores the key and the record
+func (s *memStore) Put(k string, v []byte) error {
```
Comment (troyronda) [inline pkg/storage/mem/mem_store.go]: TODO - we will need to consider automatic eviction based on TTL.

File: pkg/storage/mem/mem_store.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	if k == "" || v == nil {
+		return errors.New("key and value are mandatory")
+	}
+	s.db[k] = v
```
Comment (sandrask) [inline pkg/storage/mem/mem_store.go]: Do we need lock here for synchronization issues with map?

Comment (fqutishat) [reply pkg/storage/mem/mem_store.go]: yes

Comment (fqutishat) [reply pkg/storage/mem/mem_store.go]: i will fix it

Comment (fqutishat) [reply pkg/storage/mem/mem_store.go]: done

File: pkg/storage/mem/mem_store.go (hunk 1 above)
Comment (fqutishat) [reply pkg/storage/mem/mem_store.go]: done

File: pkg/framework/aries/framework.go [hunk 3]
```diff
@@ -24,6 +24,7 @@ import (
 type Aries struct {
 	transport                 api.TransportProviderFactory
 	storeProvider             storage.Provider
```
Comment (troyronda) [inline pkg/framework/aries/framework.go]: TODO - will we need to fix the naming (right now we have "storeProvider" and "transientStoreProvider" - a bit inconsistent).

**Pull request merged**